app/models/event.rb

```diff
@@ -75,12 +61,23 @@ def conversion_rate
     currency.conversion_rate_to_default_currency(executed_on)
   end
 
-  def this_and_previous_events
-    # ordered so that buy are before sales.
-    @this_and_previous_events ||= Event.order(:executed_on, :action)
-      .where("executed_on <= ?", executed_on)
+  def previous_event
+    @previous_event ||= Event.order(:sort_column)
+      .where("sort_column < ?", sort_column)
+      .where(user_id: user_id)
+      .where(stock_id: stock_id)
+      .last
+  end
+
+  def this_and_following_events
+    Event.order(:sort_column)
+      .where("sort_column >= ?", sort_column)
       .where(user_id: user_id)
       .where(stock_id: stock_id)
-      .includes(:currency)
+  end
+
+  def set_sort_column
+    # Ordered so that buys are before sales.
+    self.sort_column = "#{executed_on.iso8601}/#{self.class.actions[action]}/#{Time.now.to_f}"
```

Comment:
If we really want to sort buy events before the sell does the `Time.now.to_f` allows that?

Reply:
Yes, sorting of strings is from left to right

Reply:
The `Time.now.to_f` is just to have _something_ to separate the different events happening on the same date. I initially wanted to use just `#{id}`, but that of course fails on new records, because we don’t have the ID before the row is persisted.